pkg/reconciler/revision/revision.go

```diff
@@ -78,18 +94,55 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 		ServiceAccountName: rev.Spec.ServiceAccountName,
 		ImagePullSecrets:   imagePullSecrets,
 	}
-	digest, err := c.resolver.Resolve(rev.Spec.GetContainer().Image,
-		opt, cfgs.Deployment.RegistriesSkippingTagResolving)
-	if err != nil {
-		err = fmt.Errorf("failed to resolve image to digest: %w", err)
-		rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
-			v1.RevisionContainerMissingMessage(
-				rev.Spec.GetContainer().Image, err.Error()))
-		return err
-	}
 
-	rev.Status.ImageDigest = digest
+	var digestGrp errgroup.Group
+	type digestData struct {
+		digestValue        string
+		containerName      string
+		isServingContainer bool
+		image              string
+		digestError        error
+	}
 
+	digests := make(chan digestData, len(rev.Spec.Containers))
+	for _, container := range rev.Spec.Containers {
+		container := container // Standard Go concurrency pattern.
+		digestGrp.Go(func() error {
+			digest, err := c.resolver.Resolve(container.Image,
+				opt, cfgs.Deployment.RegistriesSkippingTagResolving)
+			if err != nil {
+				err = fmt.Errorf("failed to resolve image to digest: %w", err)
+				digests <- digestData{
+					image:       container.Image,
+					digestError: err,
+				}
+			} else {
+				digests <- digestData{
+					digestValue:        digest,
+					containerName:      container.Name,
+					isServingContainer: len(rev.Spec.Containers) == 1 || len(container.Ports) != 0,
+				}
+			}
+			return nil
+		})
+	}
+	digestGrp.Wait()
+	close(digests)
+	for v := range digests {
+		if v.digestError != nil {
+			rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
+				v1.RevisionContainerMissingMessage(
+					v.image, v.digestError.Error()))
+			return v.digestError
+		}
+		if v.isServingContainer {
+			rev.Status.DeprecatedImageDigest = v.digestValue
+		}
+		rev.Status.ContainerStatuses = append(rev.Status.ContainerStatuses, v1.ContainerStatuses{
```

Comment:
nit: can we make sure the order is the same as as spec.Containers?

Reply:
Hi @dprotaso I will handle this as part of separate PR as its for a visualization purpose

Reply:
I was thinking of using the container name - so the first container in the spec would be the first container in the status.

If you want to do it in another PR or create a good first issue (for someone else) feel free to unblock this PR with an `/unhold`

Reply:
Okay I have created an issue to handle that in another PR
https://github.com/knative/serving/issues/7658